lib/versionReplacer.ts

```diff
@@ -47,17 +47,28 @@ export const updateK8sYaml = (
  * @throws {Error} When the old version could not be found in the pubspec.yaml file content.
  */
 export const updatePubspecVersion = (pubspecContent: string, oldVersion: string, newVersion: string): string => {
-  const regex = `version:\\s+(${oldVersion})`;
+  const regex = `version:\\s+(${oldVersion})(?:(?:\\+)(\\d))?`;
```

Comment:
This looks like, it will only work for single numbers and not for e.g. `0.9.0+11`